mkdocs/config/config_options.py

```diff
@@ -291,6 +291,24 @@ def __init__(self, exists=False, **kwargs):
         super(FilesystemObject, self).__init__(type_=utils.string_types, **kwargs)
         self.exists = exists
 
+    def pre_validation(self, config, key_name):
+        value = config[key_name]
+
+        if not value:
+            return
+
+        if os.path.isabs(value):
+            return
+
+        if config.fname is None:
+            # Unable to determine absolute path of the config file; fall back
+            # to trusting the relative path
+            return
+
+        config_dir = os.path.dirname(config.fname)
+        value = os.path.join(config_dir, value)
+        config[key_name] = value
```

Comment:
Rather than a pre-validation method, perhaps an `abspath` method would make more sense. Then anywhere we need an absolute path, just pass the value into the `self.abspath` method. And let's put it on the `BaseConfigOption` class so its available everywhere.

Note that you missed the `custom_dir` sub-setting of the theme setting [here][1]. That is not in a `FilesystemObject` subclass. And its entirely possible that Plugins could use various of the `BaseConfigOption` sub-classes or even create their own. to validate their own sub-settings. The point is, this ability to convert a relative path to an absolute path should be available from everywhere.

[1]: https://github.com/eventray/mkdocs/blob/e7d1c70eaee033e7735a5bae3f32293e200982ee/mkdocs/config/config_options.py#L449

Reply:
And I just realized the problem with this. The config (which contains the base path) is only available from the `pre_validation` and `post_validation` methods. Therefore, we can only do this from there.